src/controllers/organisations.ts

```diff
@@ -0,0 +1,119 @@
+import { db } from '../database/db';
+import { asyncHandler } from '../middleware';
+import { cleanseData, ErrorResponse, parseSqlUpdateStmt } from '../utils';
+
+/**
+ * @desc    Get all organisations
+ * @route   GET /api/organisations
+ * @access  Public
+ */
+export const getOrganisations = asyncHandler(async (req, res) => {
+    res.status(200).json(res.advancedResults);
+});
+
+/**
+ * @desc    Get single organisation
+ * @route   GET /api/organisations/:id
+ * @access  Public
+ */
+export const getOrganisation = asyncHandler(async (req, res) => {
+    const rows = await db.one('SELECT * FROM organisations WHERE organisation_id = $1', req.params.id);
+    res.status(200).json({
+        success: true,
+        data: rows,
+    });
+});
+
+/**
+ * @desc    Create organisation
+ * @route   POST /api/organisations
+ * @access  Private
+ */
+export const createOrganisation = asyncHandler(async (req, res) => {
+    const { name, type, about, website_url, handphone, email } = req.body;
+
+    const data = {
+        name,
+        type,
+        about,
+        website_url,
+        handphone,
+        email
+    };
+
+    /**
+     * SQL Transaction and creating organisation
+     * Returns a json containing organisation data
+     */
+    const rows = await db.tx(async (query) => {
+        const createOrganisation = await query.one('INSERT INTO organisations (${this:name}) VALUES (${this:csv}) RETURNING *', data);
+        return await query.one(createOrganisation);
+    });
```

Comment:
```suggestion
const rows = await db.one('INSERT INTO organisations (${this:name}) VALUES (${this:csv}) RETURNING *', data);
        return await query.one(createOrganisation);
    });
```

This will work, I suspect this may be due to the underlying `pg-promise` library expecting a certain type of data structure to be returned. Will need to execute these queries on the db level to confirm this.